src/distributed_join.cuh

```diff
@@ -152,6 +163,219 @@ void all_to_all_comm(cudf::table_view input,
   }
 }
 
+struct compression_functor {
+  /**
+   * Compress a buffer using cascaded compression.
```

Comment:
Clarify that this functor currently synchronizes GPU execution (cannot be pipelined with other work), same for the decompressor.

Reply:
Clarified in https://github.com/rapidsai/distributed-join/pull/43/commits/e6c37662ed5d23087d6491a33563211cc2e4375b.